src/sisl/io/orca/txt.py

```diff
@@ -147,5 +147,119 @@ def read_geometry(self):
+        G["vectors"] = vectors  # g-tensor eigenvectors
+        self.readline()  # skip eigenvalue line
+        self.readline()  # skip header
+        v = self.readline().split()
+        G["eigenvalues"] = np.array(v[1:4], np.float64)
+
+        return G
+
+    @sile_fh_open()
+    def read_hyperfine_coupling(self):
+        """Reads hyperfine couplings from the EPRNMR_ATensor block
+        in the ORCA property txt file
+
+        For a nucleus :math:`k`, the hyperfine interaction is usually
+        written in terms of the symmetric :math:`3\times 3` hyperfine
+        tensor :math:`A_k` such that
+
+        .. math::
+
+           H_\text{hfi} = \boldsymbol{S} \cdot A_k \boldsymbol{I}_k
+
+        where :math:`\boldsymbol{S}_k` and :math:`\boldsymbol{I}_k`
+        represent the electron and nuclear spin operators, respectively.
+
+        For a study of hyperfine coupling in nanographenes using ORCA
+        see :cite:`Sengupta2023`.
+
+        Note
+        ----
+        The hyperfine tensors are given in units of MHz.
+
+        Returns
+        -------
+        PropertyDict or list of PropertyDict : Hyperfine coupling data (in MHz)
+        """
+        f, line = self.step_to("EPRNMR_ATensor", allow_reread=False)
+        if not f:
+            return None
+
+        def read_A():
+            A = PropertyDict()
+            # Read the EPRNMR_ATensor block
+            f, line = self.step_to("Nucleus:", allow_reread=False)
+            if not f:
+                return None
+            v = line.split()
+            A["ia"] = int(v[1])
+            A["sa"] = v[2]
+            v = self.readline().split()
+            A["isotope"] = int(v[-1])
+            v = self.readline().split()
+            A["spin"] = float(v[-1])
+            v = self.readline().split()
+            A["prefactor"] = float(v[-1])
+            self.readline()  # skip Raw line
+            self.readline()  # skip header
+            tensor = np.empty([3, 3], np.float64)
+            for i in range(3):
+                v = self.readline().split()
+                tensor[i] = v[1:4]
+            A["tensor"] = tensor  # raw A_total tensor
+            self.readline()  # skip eigenvector line
+            self.readline()  # skip header
+            vectors = np.empty([3, 3], np.float64)
+            for i in range(3):
+                v = self.readline().split()
+                vectors[i] = v[1:4]
+            A["vectors"] = vectors
+            self.readline()  # skip eigenvalue line
+            self.readline()  # skip header
+            v = self.readline().split()
+            A["eigenvalues"] = np.array(v[1:4], np.float64)  # eigenvalues of A_total
+            v = self.readline().split()
+            A["iso"] = float(v[1])  # Fermi contact A_FC
+            return A
+
+        for i in range(4):
+            line = self.readline()
+        stored_nuclei = int(line.split()[-1])
+
+        if stored_nuclei == 1:
+            return read_A()
```

Comment:
I will merge and change to always return a list. I find this more consistent, as users don't need to check in their scripts.

Reply:
Ok